arch/x86/kvm/vmx/pkvm/pkvm_host.c

```diff
@@ -75,6 +75,15 @@ static int check_pci_device_count(void)
 	return 0;
 }
 
+/*
+ * Check for the coherency of paging structures accessed through pasid table
+ * entries (in scalable mode) or context table entries (in legacy mode).
+ */
+static inline bool is_iommu_coherent(u64 ecap)
+{
+	return ecap_smts(ecap) ? !!ecap_smpwc(ecap) : !!ecap_coherent(ecap);
+}
+
```

Comment:
do we finally need make it as a public API, or just need use it in pkvm_host.c

Reply:
Possibly we will still use it just in pkvm_host.c, since once we use it to initialize `pkvm->iommu_coherent`, we just use `pkvm->iommu_coherent` value.

OTOH this function is actually the same as `iommu_paging_structure_coherency()` in `drivers/iommu/intel/iommu.c`. This is just one more example of duplication of various Intel IOMMU related definitions in pKVM and in the host IOMMU driver (other examples are context and pasid entry related functions in `arch/x86/kvm/vmx/pkvm/hyp/iommu_internal.h`). Maybe finally we'll want to remove this duplication by unifying those definitions in a common header, e.g. in `include/linux/intel-iommu.h`.

Reply:
Looks good to me, yes, finally we shall unify with common header file.